Sources/Core/Extensions/Foundation/Collection+PovioKit.swift

```diff
@@ -56,3 +158,59 @@ public extension MutableCollection {
     }
   }
 }
+
+extension MutableCollection where Self: RandomAccessCollection {
+  /// Updates the element at the specified index with a new value.
+  ///
+  /// - Parameters:
+  ///   - item: The new value to set at the specified index.
+  ///   - index: The index of the element to update.
+  /// - Returns: A boolean value indicating whether the update was successful. Returns `true` if the index is valid and the update was performed; otherwise, returns `false`.
+  ///
+  /// ## Example
+  /// ```
+  /// var array = [1, 2, 3, 4]
+  /// let success = array.update(10, at: 2)
+  /// // array is now [1, 2, 10, 4]
+  /// // success is true
+  ///
+  /// let failure = array.update(10, at: 10)
+  /// // array remains [1, 2, 10, 4]
+  /// // failure is false
+  /// ```
+  @discardableResult
+  mutating func update(_ item: Element, at index: Index) -> Bool {
```

Comment:
Would it make sense to return the updated item as Element instead of Bool? This way, nil could indicate an invalid index, potentially adding flexibility for chaining operations.

Reply:
You are passing in an item to update. I understand your notion here, but I think it doesn't make much sense tbh.